typing/ctype.ml

```diff
@@ -2683,24 +2684,33 @@ let complete_type_list ?(allow_absent=false) env fl1 lv2 pack2 =
                 if allow_absent then
                   complete nl fl2
                 else
-                  raise Exit
+                  raise (Missing n)
             end
         | (_, {type_arity = 0; type_kind = Type_abstract _;
                type_private = Public; type_manifest = None})
           when allow_absent ->
             complete nl fl2
-        | _ -> raise Exit
-        | exception Not_found when allow_absent->
+        | _ -> raise (Missing n)
```

Comment:
It is not clear to the code reader why we get a `Missing` in the catch-all case `_` here. What are those other cases? Could it be that some error cases, caused by arity mismatches, now get an incorrect error message?